resend_scheduler.go

```diff
@@ -106,28 +107,26 @@ func (rs *ResendScheduler) resendPacket(pendingPacket *PendingPacket) {
 		return
 	}
 
-	if time.Since(pendingPacket.lastSendTime) >= rs.Interval {
+	if time.Since(pendingPacket.lastSendTime) >= time.Duration(slidingWindow.streamSettings.KeepAliveTimeout) * time.Millisecond {
 		// * Resend the packet to the connection
 		server := connection.endpoint.Server
 		data := packet.Bytes()
 		server.sendRaw(connection.Socket, data)
-
-		pendingPacket.interval += rs.Increase
-		pendingPacket.ticker.Reset(pendingPacket.interval)
+		
 		pendingPacket.resendCount++
+		if (pendingPacket.resendCount < slidingWindow.streamSettings.ExtraRestransmitTimeoutTrigger) {
+			pendingPacket.interval += time.Duration(uint32(float32(slidingWindow.streamSettings.KeepAliveTimeout) * slidingWindow.streamSettings.RetransmitTimeoutMultiplier)) * time.Millisecond
+		} else {
+			pendingPacket.interval += time.Duration(uint32(float32(slidingWindow.streamSettings.KeepAliveTimeout) * slidingWindow.streamSettings.ExtraRetransmitTimeoutMultiplier)) * time.Millisecond
+		}
```

Comment:
Just a simplification:

```suggestion
		var retransmitTimeoutMultiplier uint32
		if (pendingPacket.resendCount < slidingWindow.streamSettings.ExtraRestransmitTimeoutTrigger) {
			retransmitTimeoutMultiplier = slidingWindow.streamSettings.RetransmitTimeoutMultiplier
		} else {
			retransmitTimeoutMultiplier = slidingWindow.streamSettings.ExtraRetransmitTimeoutMultiplier
		}
		pendingPacket.interval += time.Duration(uint32(float32(slidingWindow.streamSettings.KeepAliveTimeout) * retransmitTimeoutMultiplier)) * time.Millisecond
```

Reply:
I'd also like a TODO comment somewhere noting that this isn't an accurate implementation, even if it does work for our needs. We know where this is calculated at (at least in Xenoblade), we just don't fully understand how at the moment. But I'd like to do an accurate implementation at some point

Reply:
Yeah makes sense, I also thought about that.